velox/expression/EvalCtx.cpp

```diff
@@ -219,11 +219,13 @@ void EvalCtx::convertElementErrorsToTopLevelNulls(
     return;
   }
 
+  auto rawNulls = result->mutableRawNulls();
+
   const auto* rawElementToTopLevelRows =
       elementToTopLevelRows->as<vector_size_t>();
   elementRows.applyToSelected([&](auto row) {
     if (errors_->isIndexInRange(row) && !errors_->isNullAt(row)) {
-      result->setNull(rawElementToTopLevelRows[row], true);
+      bits::setNull(rawNulls, rawElementToTopLevelRows[row], true);
```

Comment:
I see your point, i did not notice that setNull actually calla ensureNulls, with the base PR this will be a perf optimization will update the summary